src/DPD-BASIC/pair_dpd_coul_slater_long.cpp

```diff
@@ -0,0 +1,573 @@
+
+ if (force->kspace == nullptr)
+    error->all(FLERR,"Pair style requires a KSpace style");
+  g_ewald = force->kspace->g_ewald;
+}
+
+/* ----------------------------------------------------------------------
+   init for one type pair i,j and corresponding j,i
+   return the maximum cutoff between Slater or DPD cutoff if charged
+   return the DPD cutoff for uncharged
+------------------------------------------------------------------------- */
+
+double PairDPDCoulSlaterLong::init_one(int i, int j)
+{
+  if (setflag[i][j] == 0) error->all(FLERR,"All pair coeffs are not set");
+
+  sigma[i][j] = sqrt(2.0*force->boltz*temperature*gamma[i][j]);
+
+  cut_dpdsq[i][j] = cut_dpd[i][j] * cut_dpd[i][j];
+  cut_dpdsq[j][i] = cut_dpdsq[i][j];
+  cut_slatersq[i][j] = cut_slater[i][j] * cut_slater[i][j];
+  cut_slatersq[j][i] = cut_slatersq[i][j];
+
+  a0[j][i] = a0[i][j];
+  gamma[j][i] = gamma[i][j];
+  sigma[j][i] = sigma[i][j];
+  scale[j][i] = scale[i][j];
+  cut_dpd[j][i] = cut_dpd[i][j];
+  cut_slater[j][i] = cut_slater[i][j];
+  cut[j][i] = cut[i][j];
+
+  //return cut[i][j];
+  return MAX(cut_dpd[i][j], cut_slater[i][j]);
+}
+
+/* ----------------------------------------------------------------------
+   proc 0 writes to restart file
+------------------------------------------------------------------------- */
+
+void PairDPDCoulSlaterLong::write_restart(FILE *fp)
+{
+  write_restart_settings(fp);
+
+  int i,j;
+  for (i = 1; i <= atom->ntypes; i++)
+    for (j = i; j <= atom->ntypes; j++) {
+      fwrite(&setflag[i][j],sizeof(int),1,fp);
+      if (setflag[i][j]) {
+        fwrite(&a0[i][j],sizeof(double),1,fp);
+        fwrite(&gamma[i][j],sizeof(double),1,fp);
+        fwrite(&cut[i][j],sizeof(double),1,fp);
+        fwrite(&cut_dpd[i][j],sizeof(double),1,fp);
+        fwrite(&cut_dpdsq[i][j],sizeof(double),1,fp);
+        fwrite(&cut_slater[i][j],sizeof(double),1,fp);
+        fwrite(&cut_slatersq[i][j],sizeof(double),1,fp);
+        fwrite(&scale[i][j],sizeof(double),1,fp);
+      }
+    }
+}
+
+/* ----------------------------------------------------------------------
+   proc 0 reads from restart file, bcasts
+------------------------------------------------------------------------- */
+
+void PairDPDCoulSlaterLong::read_restart(FILE *fp)
+{
+  read_restart_settings(fp);
+
+  allocate();
+
+  int i,j;
+  int me = comm->me;
+  for (i = 1; i <= atom->ntypes; i++)
+    for (j = i; j <= atom->ntypes; j++) {
+      if (me == 0) utils::sfread(FLERR,&setflag[i][j],sizeof(int),1,fp,nullptr,error);
+      MPI_Bcast(&setflag[i][j],1,MPI_INT,0,world);
+      if (setflag[i][j]) {
+        if (me == 0) {
+          utils::sfread(FLERR,&a0[i][j],sizeof(double),1,fp,nullptr,error);
+          utils::sfread(FLERR,&gamma[i][j],sizeof(double),1,fp,nullptr,error);
+          utils::sfread(FLERR,&cut[i][j],sizeof(double),1,fp,nullptr,error);
+          utils::sfread(FLERR, &scale[i][j],sizeof(double),1,fp, nullptr, error);
```

Comment:
Axel is correct, proc 0 needs to read in `cut_dpd[i][j]`, `cut_dpdsq[i][j]`, `cut_slater[i][j]` and `cut_slatersq[i][j]` here